File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_category/lib/src/services/category_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/category_models.dart';

/// Category service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class CategoryService {
  const CategoryService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'category';

  CategoryPageResult<Never> empty() =>
      const CategoryPageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-category/src/services/categoryService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { CategoryPageResult } from "../types/categoryModels.js";

/**
 * Category service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface CategoryService {
  readonly capability: "category";
}

export function createCategoryService(client: AppstoreAppClient): CategoryService {
  void client;
  return { capability: "category" };
}

export function emptyCategoryResult(): CategoryPageResult<never> {
  return { items: [] };
}
